comment in cte

neelasha23 opened the pull request:
## Describe your changes

Fix bug: when comments are added in the CTE query the comments should be appended after the right bracket.
This PR handles single and multiline comments.

## Issue number

Closes #886 

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)



<!-- readthedocs-preview jupysql start -->
----
:books: Documentation preview :books:: https://jupysql--894.org.readthedocs.build/en/894/

<!-- readthedocs-preview jupysql end -->

edublancas reviewed (changes requested):
there might be some edge cases here that will be difficult to discover. I think for now it's safer to just strip comments when forming the CTE.

I think a better solution is to strip out comments before storing the CTE

we already have sqlparse as a dependency, we can use this: https://sqlparse.readthedocs.io/en/latest/api/#formatting-of-sql-statements

neelasha23 commented:
> https://sqlparse.readthedocs.io/en/latest/api/#formatting-of-sql-statements

While storing the individual snippet also do we want to strip out comments?  @edublancas

edublancas commented:
> > https://sqlparse.readthedocs.io/en/latest/api/#formatting-of-sql-statements
> 
> While storing the individual snippet also do we want to strip out comments? @edublancas

yes, when storing

neelasha23 commented:
I have made the change using strip_comments, however the CI is failing because https://docs.cloud.ploomber.io/en/latest/dashboards/jupyterlab-plugin.html is broken

edublancas commented on CHANGELOG.md:
```diff
@@ -3,6 +3,7 @@
 ## 0.10.3dev
 
 * [Fix] Remove force deleted snippets from dependent snippet's `with` (#717)
+* [Fix] Comments added in SQL query to be appended correctly when forming CTE (#886)
```
is this accurate? I think we're now deleting comments before storing snippets

neelasha23 replied:
[Fixed](https://github.com/ploomber/jupysql/pull/894/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR6)

edublancas commented on src/tests/test_magic_cte.py:
```diff
@@ -182,3 +182,124 @@ def test_query_syntax_error(ip):
     assert CTE_MSG.strip() in str(excinfo.value)
+
+
+def test_comment_in_query_added_after_bracket(ip):
```
can you add more clearer test name? I believe what we're testing is that the stored snippets do not contain comments

neelasha23 replied:
[fixed](https://github.com/ploomber/jupysql/pull/894/commits/104713b221173966cf14c3a25e66a42390dd5724)

edublancas commented on src/sql/store.py:
```diff
@@ -91,6 +92,7 @@ def store(self, key, query, with_=None):
                 f"Script name ({key!r}) cannot appear in with_ argument"
             )
 
+        query = sqlparse.format(query, strip_comments=True)
```
add some comment explaining why we're removing comments

neelasha23 replied:
[fixed](https://github.com/ploomber/jupysql/pull/894/files#diff-a3d1291b54f0d59f191101a81a116bad5859a2bd81603b9b7c1cbea77d18cc10R94-R97)

edublancas commented:
@neelasha23 please update the link so the CI passes: https://docs.cloud.ploomber.io/en/latest/apps/jupyterlab-plugin.html

neelasha23 commented:
Addressed the comments. It looks the the `ipython` release yesterday is causing some tests to fail. Have pinned for now and opened an issue : https://github.com/ploomber/jupysql/issues/903
